pgx/src/lib.rs

```diff
@@ -319,4 +352,12 @@ macro_rules! pg_sql_graph_magic {
 #[allow(unused)]
 pub fn initialize() {
     pg_sys::panic::register_pg_guard_panic_hook();
+    match Lazy::force(&UTF8DATABASE) {
+        Utf8Compat::Maybe => notice!("database encoding is not validated as UTF-8"),
+        Utf8Compat::Ascii => {
+            warning!("database encoding is not UTF-8, extension will accept ASCII only")
+        }
+        Utf8Compat::No => unreachable!("unsupported database encoding encountered"),
```

Comment:
Hm. Maybe I should just remove this variant and bail up-front, when creating the enum. I'll come back to this on Monday.